extensions/sql-migration/src/wizard/assessmentDetailsPage/instanceSummary.ts

```diff
@@ -0,0 +1,354 @@
+import { ConfigDialogSetting, TdeMigrationModel } from '../../models/tdeModels';
+import { TelemetryAction, TelemetryViews, getTelemetryProps, sendSqlMigrationActionEvent } from '../../telemetry';
+
+interface IActionMetadata {
+	label: string,
+	count?: number,
+	iconPath?: azdata.IconPath,
+	color?: string
+}
+
+// Class that defines Instance Summary section.
+export class InstanceSummary {
+	private _view!: azdata.ModelView;
+	private _assessedDatabases!: azdata.TextComponent;
+	private _totalFindingLabels!: azdata.TextComponent;
+	private _yAxisLabels: azdata.TextComponent[] = [];
+	private _divs: azdata.FlexContainer[] = [];
+	private _disposables: vscode.Disposable[] = [];
+	private _tdeInfoContainer!: azdata.FlexContainer;
+	private _tdeConfigurationDialog!: TdeConfigurationDialog;
+	private _previousMiTdeMigrationConfig: TdeMigrationModel = new TdeMigrationModel(); // avoid null checks
+	private _tdeDatabaseSelectedHelperText!: azdata.TextComponent;
+	private _tdeEditButton!: azdata.ButtonComponent;
+
+	constructor(public migrationStateModel: MigrationStateModel) { }
+
+	// function to create instance summary components
+	public async createInstanceSummaryContainerAsync(view: azdata.ModelView): Promise<azdata.FlexContainer> {
+		this._view = view;
+		const instanceSummaryContainer = view.modelBuilder.flexContainer().withLayout({
+			flexFlow: 'column'
+		}).withProps({
+			CSSStyles: {
+				'padding-left': '10px'
+			}
+		}).component();
+
+		const description = view.modelBuilder.text().withProps({
+			value: constants.READINESS_SECTION_TITLE,
+			CSSStyles: {
+				...styles.SUBTITLE_LABEL_CSS,
+				'margin': '0px'
+			}
+		}).component();
+
+		this._assessedDatabases = view.modelBuilder.text().withProps({
+			value: "",
+			CSSStyles: {
+				...styles.LIGHT_LABEL_CSS,
+				'margin': '0px'
+			}
+		}).component();
+
+		const findingsSummaryTitle = view.modelBuilder.text().withProps({
+			value: constants.INSTANCE_FINDING_SUMMARY,
+			CSSStyles: {
+				...styles.SUBTITLE_LABEL_CSS,
+				'margin': '0px',
+				'margin-top': '30px'
+			}
+		}).component();
+
+		this._totalFindingLabels = view.modelBuilder.text().withProps({
+			value: "",
+			CSSStyles: {
+				...styles.LIGHT_LABEL_CSS,
+				'margin': '0px'
+			}
+		}).component();
+
+		this._tdeInfoContainer = await this.createTdeInfoContainerAsync();
+
+		instanceSummaryContainer.addItems([description,
+			this._assessedDatabases, this.createGraphComponent(),
+			this._tdeInfoContainer, findingsSummaryTitle,
+			this._totalFindingLabels]);
+
+		return instanceSummaryContainer;
+	}
+
+	// function to populate instance summary container with latest values.
+	public async populateInstanceSummaryContainerAsync(): Promise<void> {
+		this._previousMiTdeMigrationConfig = this.migrationStateModel.tdeMigrationConfig;
+		await this.refreshTdeViewAsync();
+		this._assessedDatabases.value = constants.ASSESSED_DBS_LABEL + ": " + this.migrationStateModel._databasesForAssessment?.length;
```

Comment:
same here and following.  please create a method in constants that creates the entire string and handles databasesforassessment.length as a parameter

Reply:
done